includes/api.inc

```diff
@@ -38,14 +38,20 @@ function islandora_bookmark_get_user_owned_bookmarks($type = NULL) {
 
   // User is anon, use session as well.
   if ($user->uid == 0) {
-    if (!isset($_SESSION) || empty($_SESSION['islandora_bookmark']) && variable_get('islandora_bookmark_create_user_default_lists', FALSE)) {
-      BookmarkDefaultSessionList::createNewList(t('My Default List'), 'bookmark_default', $user);
+    if (!isset($_SESSION) || !isset($_SESSION['islandora_bookmark']) || empty($_SESSION['islandora_bookmark']) && variable_get('islandora_bookmark_create_user_default_lists', FALSE)) {
```

Comment:
@jonathangreen i'm looking at the logic right now. 
According to myself: 
```PHP
if (
  ( // Open "("
   !isset($_SESSION) || // No SESSION
   !isset($_SESSION['islandora_bookmark']) ||  // OR at least not a SESSION with an islandora_bookmark entry 
   empty($_SESSION['islandora_bookmark'])  //  OR it is there but it is empty 
  )//close ")"
&& variable_get('islandora_bookmark_create_user_default_lists', FALSE))  // and create default list is TRUE
````
So adding extra () around the session conditionals should be ok?. Which would mean in human language:
If all the session stuff is empty AND i´m asked to create a default list (evaluated to TRUE), then do it.

I wonder if you would be ok if I move all the anonymous user functionality directly into cookies? The original patch was to avoid the issue in a less disruptive way (seems like I had no luck in that department! but I read that there are a lot of expectations for anonymous user bookmark lists management. Like renaming and adding new ones. It's a bit novel for me since that type of behavior is normally reserved to logged in users in other systems, really can't find right now any other system hat allows so much freedom.